mne/tests/test_epochs.py

```diff
@@ -1476,30 +1475,31 @@ def test_epochs_io_preload(tmp_path, preload):
     assert_equal(epochs.get_data().shape[-1], 1)
 
 
-@pytest.mark.parametrize(
-    "split_size, n_epochs, n_files, size",
-    [
-        ("1.5MB", 9, 6, 1572864),
-        ("3MB", 18, 3, 3 * 1024 * 1024),
-    ],
-)
-@pytest.mark.parametrize("metadata", [False, True])
-@pytest.mark.parametrize("concat", (False, True))
-def test_split_saving(tmp_path, split_size, n_epochs, n_files, size, metadata, concat):
-    """Test saving split epochs."""
-    if metadata:
+@pytest.fixture(params=[True, False], ids=["metadata", "no_metadata"])
+def is_add_metadata(request):
+    """Whether to create metadata in epochs fixture."""
+    return request.param
+
+
+@pytest.fixture(params=[True, False], ids=["concat", "no_concat"])
+def is_concat(request):
+    """Whether to create concatenated version of epochs."""
+    return request.param
+
+
+@pytest.fixture
+def fake_epochs(n_epochs, is_add_metadata, is_concat):
+    """Epochs object to test split naming."""
+    n_ch, fs = 100, 1000.0
+    if is_add_metadata:
         pytest.importorskip("pandas")
     # See gh-5102
-    fs = 1000.0
     n_times = int(round(fs * (n_epochs + 1)))
-    raw = mne.io.RawArray(
-        np.random.RandomState(0).randn(100, n_times), mne.create_info(100, 1000.0)
-    )
+    raw_data = np.random.RandomState(0).randn(n_ch, n_times)
+    raw = mne.io.RawArray(raw_data, mne.create_info(n_ch, fs))
     events = mne.make_fixed_length_events(raw, 1)
     epochs = mne.Epochs(raw, events)
-    if split_size == "2MB" and (metadata or concat):
-        n_files += 1
```

Comment:
Not sure what this was doing. Removed it.

Reply:
This was a special case inside the test. Basically it was necessary because in the case where your `split_size` is small and you have metadata (or concat, for whatever reason), you will end up with another file. It doesn't seem like your PR should change the file sizes produced but rather just the filenames, so it seems like a bug (either on this PR or in `main` currently) that you were able to remove this and have things still pass.

Reply:
... looks like it is a bug on `main` because we never use `split_size="2MB"`, so removing this seems okay